consensus/polybft/consensus_runtime.go

```diff
@@ -738,20 +752,18 @@ func (c *consensusRuntime) InsertBlock(proposal []byte, committedSeals []*messag
 		return
 	}
 
-	if c.IsBridgeEnabled() {
-		if (fsm.isEndOfEpoch || c.checkpointManager.isCheckpointBlock(fullBlock.Block.Header.Number)) &&
-			bytes.Equal(c.config.Key.Address().Bytes(), fullBlock.Block.Header.Miner) {
-			go func(header types.Header, epochNumber uint64) {
-				if err := c.checkpointManager.submitCheckpoint(header, fsm.isEndOfEpoch); err != nil {
-					c.logger.Warn("failed to submit checkpoint",
-						"checkpoint block", header.Number,
-						"epoch number", epochNumber,
-						"error", err)
-				}
-			}(*fullBlock.Block.Header, fsm.epochNumber)
-
-			c.checkpointManager.latestCheckpointID = fullBlock.Block.Number()
-		}
+	if c.checkpointManager.IsCheckpointBlock(fullBlock.Block.Header.Number, fsm.isEndOfEpoch) &&
+		bytes.Equal(c.config.Key.Address().Bytes(), fullBlock.Block.Header.Miner) {
+		go func(header *types.Header, epochNumber uint64) {
+			if err := c.checkpointManager.SubmitCheckpoint(header, fsm.isEndOfEpoch); err != nil {
+				c.logger.Warn("failed to submit checkpoint",
+					"checkpoint block", header.Number,
+					"epoch number", epochNumber,
+					"error", err)
+			}
+		}(fullBlock.Block.Header, fsm.epochNumber)
+
+		c.checkpointManager.SetLatestCheckpointID(fullBlock.Block.Number())
```

Comment:
Is it ok to update latest checkpoint id even if submit checkpoint fails (which is called from another routine)?

Reply:
Good catch. No, we should not.

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1113/commits/be68a84e56c50260e2d3fcf66bcc559212cc3375